src/observer.ts

```diff
@@ -88,19 +105,27 @@ function getArnsResolution({
     });
 
     stream.on('data', (data) => {
-      dataHash.update(data);
+      const bytesToProcess = Math.min(
+        data.length,
+        MAX_BYTES_TO_PROCESS - streamBytesProcessed,
+      );
+
+      if (bytesToProcess > 0) {
+        dataHash.update(data.slice(0, bytesToProcess));
+        streamBytesProcessed += bytesToProcess;
+      }
+
+      if (streamBytesProcessed >= MAX_BYTES_TO_PROCESS) {
+        stream.on('close', () => {
+          resolveWithResponse(response);
+        });
+
+        stream.destroy();
```

Comment:
@dtfiedler @djwhitt I had to listen to `close` here as `stream.destroy()` will close the response so `end` will never be emitted.

I listen to this only inside this if so we don't resolve twice as `got` will emit `close` AND `end` when a request ends normally (without empirically destroying it)

Reply:
Nah, this seems fine to me. You could make the event handler more clever instead, but I think that just obscures what's actually happening.